src/sesclient.native/sesclient.native.cpp

```diff
@@ -43,62 +49,172 @@ struct Initializer
     }
 };
 
+struct ParameterParser
+{
+public:
+
+    void parse(int argc, char** argv)
+    {
+        if ((argc % 2) == 1)
+        {
+            // We have pairs of parameters, collect them into our map
+            while (argc != 1)
+            {
+                auto value = argv[--argc];
+                auto key = argv[--argc];
+                _parameters.emplace(key, value);
+            }
+        }
+
+        if (_parameters.size() == 0)
+        {
+            _shouldShowUsage = true;
+        }
+        else
+        {
+            checkForMandatoryParameters(_parameters);
+            checkForExtraParameters(_parameters);
+        }
+    }
+
+    bool contains(std::string name)
```

Comment:
1. pass strings by `const std::string&` to avoid copying them
2. this method can be `const` (goes on the end, after parentheses)